### PR TITLE
Add support for exec based auth.

### LINKED ISSUE #34: Port exec-based client auth provider to Javascript client
With 1.10, this exec-based client auth credential provider alpha feature was introduced  [kubernetes/features#541](https://github.com/kubernetes/features/issues/541) for client-go.  Exec-based auth plugin should be supported by this Javascript client in addition to client-go. Umbrella issue: [kubernetes#62185](https://github.com/kubernetes/kubernetes/issues/62185)

/kind feature
/sig api-machinery

### PR DESCRIPTION
**Pull request by brendandburns**
Closes #34

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -3,17 +3,21 @@ import https = require('https');
...
-import jsonpath = require('jsonpath/jsonpath.min');
 import request = require('request');
 import shelljs = require('shelljs');
 
-import api = require('./api');
+import * as api from './api';
+import { Authenticator } from './auth';
 import { Cluster, Context, newClusters, newContexts, newUsers, User } from './config_types';
+import { CloudAuth } from './cloud_auth';
+import { ExecAuth } from './exec_auth';
```
[the remark this answers is not in the document]
Comment (brendandburns) [reply src/config.ts]: They're right there on my ....
_checks notes_
... my local file system.

*sigh*

Fixed.

**Comment by brendandburns** [comment]
Fixed lint errors....

File: src/config.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (drubin) [reply src/config.ts]: Thanks! and thanks again for adding this @brendanburns

**Comment by SawyerHopkins** [comment]
I am using aws-iam-authenticator and the response body has the shape

```json
{ 
  "kind": "ExecCredential",
  "apiVersion": "client.authentication.k8s.io/v1alpha1",
  "spec": {},
  "status": { 
    "expirationTimestamp": "2018-12-21T23:29:01Z",
    "token": "<REDACTED>"
  }
}
```

Which appears correct according to https://kubernetes.io/docs/reference/access-authn-authz/authentication/

This however causes this line to return an undefined value for the bearer token. https://github.com/kubernetes-client/javascript/pull/158/files#diff-d03480d634d6f56466d9e8bf3e1b00d0R32

I have a locally linked build which returns `return Bearer ${obj.status.token}` which appears to work correctly.

Is this an issue with `aws-iam-authenticator`, is this because the auth api version is still v1alpha1, or have I setup something incorrectly in my kubectl?

```yaml
users:
  - name: <REDACTED>
    user:
      auth-provider:
        name: exec
        config:
          exec:
            apiVersion: client.authentication.k8s.io/v1alpha1
            command: aws-iam-authenticator
            args:
              - token
              - -i
              - <REDACTED>
      exec:
        apiVersion: client.authentication.k8s.io/v1alpha1
        command: aws-iam-authenticator
        args:
          - token
          - -i
          - <REDACTED>
```

**Comment by brendandburns** [comment]
@SawyerHopkins I think this is just a straight-up bug, I didn't have a working kubeconfig to test against and I messed up....

I'll send a fix asap.

**Comment by ndeloof** [comment]
afaict this change broke the fix introduced by https://github.com/kubernetes-client/javascript/pull/132 to support electron.
shelljs isn't compatible with Electron without known possible fix yet https://github.com/shelljs/shelljs/wiki/Electron-compatibility

**Pull request merged**